### PR TITLE
[JENKINS-42744] - Prevent conversion of environment variables to lowercase

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change intends fixing the regression I've introduced in WinSW 2.0 in https://github.com/kohsuke/winsw/commit/92367f44ef208bca20c6dfcadb975722fd28763d. I have added `ps.EnvironmentVariables[WinSWSystem.ENVVAR_NAME_SERVICE_ID.ToLower()] = _descriptor.Id;`, which made all variables to be in lowercase in the injected process. Some tools were not ready to that though they are expected to handle Env Vars in the case-insensitive mode in Windows. But Java tools do not do that By default, hence [JENKINS-42744](https://issues.jenkins-ci.org/browse/JENKINS-42744) was reported by Jenkins users.

Changes:

- [x] - Fix Issue #201 (JENKINS-42744)
- [x] - Refactor the `Main#StartProcess()` logic to make it testable
- [x] - Add tests for #201 
- [x] - Also add the process termination tests to the `RunawayProcessKiller` to ensure there is no regressions

https://issues.jenkins-ci.org/browse/JENKINS-42744

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@jglick I have splitted the chanve to multiple commits for a reason. Here is the actual fix: https://github.com/kohsuke/winsw/pull/202/commits/f81f5d3c574d6b72f08a9325845c25541e6c34ab

**Comment by oleg-nenashev** [comment]
@reviewbybees done

File: src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs [hunk 1]
```diff
@@ -38,9 +44,12 @@ public RunawayProcessKillerExtension()
             // Default initializer
         }
 
-        public RunawayProcessKillerExtension(String pidfile)
+        public RunawayProcessKillerExtension(String pidfile, int stopTimeoutMs = 5000, bool stopParentFirst = false, bool checkWinSWEnvironmentVariable = true)
         {
             this.Pidfile = pidfile;
+            this.StopTimeout = TimeSpan.FromMilliseconds(5000);
```
Comment (oleg-nenashev) [inline src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs]: This is a bug, but this API is not being used anywhere outside tests. will fix in a follow-up

File: src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs [hunk 2]
```diff
@@ -38,9 +38,11 @@ public RunawayProcessKillerExtension()
             // Default initializer
         }
 
-        public RunawayProcessKillerExtension(String pidfile)
+        public RunawayProcessKillerExtension(String pidfile, int stopTimeoutMs = 5000, bool stopParentFirst = false)
```
Comment (daniel-beck) [inline src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs]: Unused argument

File: src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs [hunk 3]
```diff
@@ -113,11 +116,12 @@ public override void OnWrapperStarted()
             else
             {
                 Logger.Warn("The process " + pid + " has no " + expectedEnvVarName + " environment variable defined. " 
-                    + "The process has not been started by this service, hence it won't be terminated.");
+                    + "The process has not been started by WinSW, hence it won't be terminated.");
```
Comment (daniel-beck) [inline src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs]: Isn't it `jenkins.exe` for Jenkins? Is it reasonable to expect users to know what WinSW is where this message will show up?

Comment (oleg-nenashev) [inline src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs]: WinSW is not for Jenkins. It is a standalone project, which is being widely used outside. But I agree the message is not ideal. The intention was to say that "The process has not been created by any of winsw-powered services you may have on this machine", but maybe it is better to revert the change

File: src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs (hunk 2 above)
Comment (oleg-nenashev) [inline src/Plugins/RunawayProcessKiller/RunawayProcessKillerExtension.cs]: Yep, commented in the PR. Will fix in the follow-up though it is harmless in the current version (WinSW API is for internal use only)
